tests/test_consumer.py

```
@@ -346,3 +346,291 @@ async def test_consumer_session_cleanup_on_connection_failure(self):
+            assert "active_shards" in status
+            assert "closed_shards" in status
+            assert "allocated_shards" in status
+            assert "shard_details" in status
+            
+            # Should have at least one shard for the test stream
+            assert status["total_shards"] >= 1
+
+    @pytest.mark.asyncio
+    async def test_consumer_closed_shard_handling(self, endpoint_url):
+        """Test handling of closed shards."""
+        from unittest.mock import AsyncMock, patch
```

Ruff (0.11.9)

Redefinition of unused `AsyncMock` from line 3

Remove definition: `AsyncMock`

(F811)